### PR TITLE
Increase test coverage for db/backends/base/operations.py.

### PR DESCRIPTION
**Pull request by atombrella**

### CODE REVIEW

File: tests/backends/base/test_operations.py [hunk 1]
```diff
@@ -1,8 +1,62 @@
...
+            self.base.sql_flush(None, None, None)
+
+    @skipIfDBFeature('supports_temporal_subtraction')
+    def test_subtract_temporals(self):
+        duration_field_internal_type = DurationField.get_internal_type()
+        msg = (
+            'This backend does not support %s subtraction.' %
+            duration_field_internal_type
+        )
+        with self.assertRaisesMessage(NotImplementedError, msg):
```
Comment (atombrella) [inline tests/backends/base/test_operations.py]: According to https://www.python.org/dev/peps/pep-0249/ this should actually be `NotSupportedError`. Can I change it in the same PR? The same applies to `NotImplementedError` in `distinct_sql`.

Comment (timgraham) [reply tests/backends/base/test_operations.py]: You can change it in a separate commit in the same PR.

Comment (atombrella) [reply tests/backends/base/test_operations.py]: I made the change the first commit in this PR. I'm not sure about the commit message (mainly the header).

File: tests/backends/base/test_operations.py [hunk 2]
```diff
@@ -1,8 +1,60 @@
 from django.db import NotSupportedError, connection
+from django.db.backends.base.operations import BaseDatabaseOperations
+from django.db.models import DurationField
 from django.test import SimpleTestCase, skipIfDBFeature
 
 
 class DatabaseOperationTests(SimpleTestCase):
+    def setUp(self):
+        self.ops = BaseDatabaseOperations(connection=connection)
```
Comment (timgraham) [inline tests/backends/base/test_operations.py]: Is there a difference between using `BaseDatabaseOperations(connection=connection)` and `connection.ops`? I guess we should use one style consistently.

Comment (atombrella) [reply tests/backends/base/test_operations.py]: In this case, yes. `connection.ops` is the instantiated `ops_class` from `DatabaseWrapper` and won't be this base class for an actual connection. `@override_settings(DATABASES=None)` sets it to the dummy backend, but `quote_name` has a different value for the operations there.

**Pull request merged**